values/binary_test.go

```diff
@@ -552,8 +557,13 @@ func TestBinaryOperator(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
-
-			if want, got := Value(tt.want), fn(left, right); !ValueEqual(want, got) {
+			want := Value(tt.want)
+			got, err := fn(left, right)
+			if tt.wantErr != nil {
+				if err == nil {
```

Comment:
This logic doesn't look quite right yet.  If `fn` returns an error and one wasn't expected, then the test should fail with a message like `unexpected error: <errormsg>`.